PL2/PL2-gui/src/main/java/nl/tudelft/pl2016gr2/gui/view/RootLayoutController.java

```diff
@@ -102,24 +188,54 @@ private void initializeGraphIcon() {
         return;
       }
       mainPane.getItems().clear();
-      mainPane.getItems().add(graphPane);
+      mainPane.getItems().add(drawGraphs.getGraphPane());
       zoomOutButtonDisabled = zoomOutButton.isDisabled();
       zoomOutButton.setDisable(true);
       mainPane.fireEvent(new GraphicsChangedEvent());
     });
   }
 
   /**
-   * Set the data which has to be visualized.
-   *
-   * @param root the root of the tree which has to be drawn.
+   * Thread which is used to get a graph of a subset of the genomes from a graph.
    */
-  public void setData(IPhylogeneticTreeNode root) {
-    assert treeManager == null;
-    treeManager = new TreeManager(treePane, root, zoomOutButton, selectionManager);
-    heatmapManager.initLeaves(treeManager.getCurrentLeaves());
-    treeManager.setOnLeavesChanged((Observable observable, Object arg) -> {
-      heatmapManager.setLeaves(treeManager.getCurrentLeaves());
-    });
+  private class SplitGraphsThread extends Thread {
```

Comment:
This class is only used by the RootLayoutController class and is only used to execute a single method call on a separate thead. If I wouldn't have multithreaded, this would simply have been 1 method call, so in my opinion this class belongs within the this class.